presto-hive/src/main/java/com/facebook/presto/hive/pagefile/PageFilePageSourceFactory.java

```diff
@@ -0,0 +1,114 @@
+import static com.google.common.base.Strings.nullToEmpty;
+import static com.google.common.base.Throwables.propagateIfPossible;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class PageFilePageSourceFactory
+        implements HiveBatchPageSourceFactory
+{
+    private final HdfsEnvironment hdfsEnvironment;
+    private final BlockEncodingSerde blockEncodingSerde;
+
+    @Inject
+    public PageFilePageSourceFactory(
+            HdfsEnvironment hdfsEnvironment,
+            BlockEncodingSerde blockEncodingSerde)
+    {
+        this.hdfsEnvironment = requireNonNull(hdfsEnvironment, "hdfsEnvironment is null");
+        this.blockEncodingSerde = requireNonNull(blockEncodingSerde, "blockEncodingSerde is null");
+    }
+
+    @Override
+    public Optional<? extends ConnectorPageSource> createPageSource(
+            Configuration configuration,
+            ConnectorSession session,
+            Path path,
+            long start,
+            long length,
+            long fileSize,
+            Storage storage,
+            SchemaTableName tableName,
+            Map<String, String> tableParameters,
+            List<HiveColumnHandle> columns,
+            TupleDomain<HiveColumnHandle> effectivePredicate,
+            DateTimeZone hiveStorageTimeZone,
+            HiveFileContext hiveFileContext,
+            Optional<EncryptionInformation> encryptionInformation)
+    {
+        if (!PageInputFormat.class.getSimpleName().equals(storage.getStorageFormat().getInputFormat())) {
+            return Optional.empty();
+        }
+
+        FSDataInputStream inputStream;
+        try {
+            inputStream = hdfsEnvironment.getFileSystem(session.getUser(), path, configuration).openFile(path, hiveFileContext);
+        }
+        catch (Exception e) {
+            if (nullToEmpty(e.getMessage()).trim().equals("Filesystem closed") ||
+                    e instanceof FileNotFoundException) {
+                throw new PrestoException(HIVE_CANNOT_OPEN_SPLIT, e);
+            }
+            throw new PrestoException(HIVE_CANNOT_OPEN_SPLIT, splitError(e, path, start, length), e);
+        }
+
+        try {
+            PageFilePageSource pageFilePageSource = new PageFilePageSource(inputStream, start, length, fileSize, blockEncodingSerde, columns);
+            return Optional.of(pageFilePageSource);
+        }
+        catch (Throwable e) {
+            try {
+                inputStream.close();
+            }
+            catch (IOException ignored) {
+            }
+            propagateIfPossible(e, PrestoException.class);
+            throw new PrestoException(HIVE_CANNOT_OPEN_SPLIT, splitError(e, path, start, length), e);
+        }
+    }
+
+    private static String splitError(Throwable t, Path path, long start, long length)
+    {
+        return format("Error opening Hive split %s (offset=%s, length=%s): %s", path, start, length, t.getMessage());
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code uses &#39;%s&#39; to format long: start (declared at line 110), length (declared at line 110). This is a potential locale-sensitive handling issue. It might cause errors in the handling and processing of the statement at line: 112. Consider formatting this data with &#39;%d&#39; instead.